src/bot/MessageHandler.ts

```diff
@@ -34,15 +34,33 @@ export default class MessageHandler {
     );
   }
 
-  private execute(message: Message) {
+  private async execute(message: Message) {
     const [command, ...params] = message.content.split(' ');
     const commandToRun = this.commands.get(command);
 
     if (commandToRun.elevated && !userHasElevatedRole(message.member)) {
-      message.channel.send(localize.t('errors.unauthorized'));
+      await message.channel.send(localize.t('errors.unauthorized'));
       return;
     }
 
-    commandToRun.run(message, params);
+    const statusMsg = await message.reply(localize.t('status.received', { command }));
+    statusMsg.reference = {
+      // This should really be done by the discordjs library since we are replying
+      channelID: message.channel.id,
+      guildID: message.guild?.id || '',
+      messageID: message.id
+    };
```

Comment:
We potentially do not need this. Would there speak anything against editing the actual message of the user directly instead of an additional status message?

Since the bot says `Running {command}` information about which command was executed would not be lost.